src/framework/varQuery.ml

```diff
@@ -0,0 +1,6 @@
+type t =
+  | Global of CilType.Varinfo.t
+  | Node of {node: Node.t; fundec: CilType.Fundec.t option} (** Optional [fundec] override to allow querying old state in incremental. *)
+[@@deriving ord]
```

Comment:
What exactly is the meaning or use case of the `fundec` here? I.e. how does the `fundec` help in querying an old states of a node?

Reply:
Without it, the way it's implemented in `FromSpec.iter_vars` is to take the given node, find it's `fundec` using `Node.find_fundec` and then look up the contexts for that function.

The problem in the incremental case is that `Node.find_fundec` doesn't do the lookup in the old AST, but the new one (by default in `Cilfacade`), so it would fail to find any old contexts for the old changed function.